### DISCUSSION TITLE
Obtaining ExchangeId & IndexId

### DISCUSSION THREAD

**Discussion by derhallim** [pnp/pnpframework#126] [Q&A] [answered]
When exporting the template for my pages, the Events Web Part has entries for ExchangeId and IndexId.. I'd like to deploy multiple events web parts on the same page, each pointing to a different site. How do we deal with the IndexId and ExchangeId that are available in the exported code? For the ExchangeId there's a question about that on Stackexchange that hasn't been answered yet, but the ExchangeId is a combination of SiteId, WebId and some other values that are not clear to me.

**Comment by jansenbe** [comment]
You can try to reverse engineer, e.g. put the events web part of the site you want get events from , check the settings in the web part json config (tip, add the `maintenancemode=true` url parameter when loading the page with web part) and then reuse those. 

You could also fiddle with the web part json and see if the web part also works without providing this exchangeId value

**Comment by derhallim** [comment]
@jansenbe I want to deploy all of these sites at the same time, to have a reusable solution where I provision 5 sites for example, and then get their events from a 6th site that I deploy along with them. So I won't know the IndexId or ExchangeId. 

Is there a way to get these values programmatically? I can store these in a variable then set it on the Events web part in my main site.. however I'm not sure how to get these values.

**Comment by jansenbe** [comment, marked as answer]
There's no public information on these values, only option I can give it to see if the web part works without setting these values: e.g. open Fiddler, put the web part on the page and save the page. Check which request sends the page to the server and see how the web part json looks: might be that the value is set server side based upon the information provided. Given this is not a bug/issue I'm going to transfer this one into the discussions list.
